### PR TITLE
Extend MembersThat and MembersShould to have starting,… (Resolves #239)

### LINKED ISSUE #239: consistent methods for classes() and methods()
`classes().that().haveSimpleNameStartingWith()` is available but `methods().that().haveSimpleNameStartingWith()` is not.

we need to use `methods().that().haveNameMatching()`

wouldn't it be nice to have the methods of `classes()` and `methods()` as consistent as possible? or is there a specific reason not to?

### PR DESCRIPTION
**Pull request by kamer**
… containing and ending functionality.

It's my first attempt to contribute an open-source project. 
Hope this solves the issue properly.

Resolves: #239

### CODE REVIEW

**Comment by hankem** [comment]
Cool, thanks! 😉
As I was curious, I had a first quick look and two tiny comments.

File: archunit/src/main/java/com/tngtech/archunit/lang/syntax/elements/MembersShould.java [hunk 1]
```diff
@@ -107,6 +107,33 @@
     @PublicAPI(usage = ACCESS)
     CONJUNCTION haveFullNameNotMatching(String regex);
 
+    /**
+     * Asserts that members have a full name starting with given prefix.
```
Comment (hankem) [inline archunit/src/main/java/com/tngtech/archunit/lang/syntax/elements/MembersShould.java]: If I'm not mistaken, the new methods test member _names_, but not _full names_ (which would include their owners' FQCN).

File: archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/GivenMembersTest.java [hunk 2]
```diff
@@ -214,6 +214,23 @@ public void complex_members_syntax() {
...
+                $(described(codeUnits().that().haveNameContaining("t")), ALL_CODE_UNIT_DESCRIPTIONS),
+                $(described(methods().that().haveNameContaining("dA")), ImmutableSet.of(METHOD_A)),
+                $(described(constructors().that().haveNameContaining("init")), ALL_CONSTRUCTOR_DESCRIPTIONS),
+                $(described(fields().that().haveNameContaining("ie")), ALL_FIELD_DESCRIPTIONS),
+                $(described(members().that().haveNameEndingWith("D")), ImmutableSet.of(FIELD_D, METHOD_D)),
+                $(described(codeUnits().that().haveNameEndingWith("A")), ImmutableSet.of(METHOD_A)),
+                $(described(methods().that().haveNameEndingWith("C")), ImmutableSet.of(METHOD_C)),
+                $(described(constructors().that().haveNameEndingWith("it>")), ALL_CONSTRUCTOR_DESCRIPTIONS),
+                $(described(fields().that().haveNameEndingWith("B")), ImmutableSet.of(FIELD_B)),
+
```
Comment (hankem) [inline archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/GivenMembersTest.java]: We could probably add similar test coverage for the _members should_ functinality in `MembersShouldTest`.

**Comment by kamer** [comment]
Update: 
- Comments in MembersShould is changed since new methods test _name_ instead of _fullName_.
- Tests added in `MembersShouldTest`.

**Comment by codecholeric** [comment]
One thing I noticed though (out of scope for this PR) is that we really should look into this zoo of `ArchConditions` that really only differ in this little this of how to create the positive or negative description of a failure. Sometimes we have `satisfied ? predicate.getDescription().replace(..) : ...` sometimes it's hard-coded in the description and duplicated, just seems to be a lot of different ways and duplication just to handle this details creation. Maybe `DescribedPredicate` should be extended a little so the description is more sophisticated and can handle different grammatical configurations...

**Comment by codecholeric** [comment]
@kamer Any update?

**Comment by kamer** [comment]
@codecholeric  I updated like you have described and build with `./gradlew clean build -PallTests` as it is recommended in contributing guide. I built successfully locally but build checks fails. Am I doing something wrong?

**Comment by codecholeric** [comment]
@kamer thanks for cleaning it up!! :smiley: When I tested the branch locally the tests also didn't pass, because of changed logic with respect to the line number -> #344
I've fixed it real quick. (I touched those tests anyway in my last commit, so it was a simple one line fixup in the assertion)

I adjusted the commit message of your commit and removed all the blubber from my review commits, because I do not think these comments will provide much benefit (adjust javadoc, etc.), when I read this history entry in 6 months :wink: (you can adjust the message, when you squash commits, you don't have to append them all to one long sequence)
Also I noticed that you did a merge `master` -> `feature-branch`, I normally always try to do a rebase + force push (i.e. `git pull --rebase upstream master` + `git push --force-with-lease`). The consequence is, that the history will stay linear when you merge back into `master`, which I highly appreciate. I've adjusted this on your branch, so instead of the merge commit, the branch is now rebased on upstream master

Anyway, thank you so much for your support!! Gonna merge it now  :smiley:

**Pull request merged**